File: java/src/main/java/api/jdy/FormDataDemo.java
package api.jdy;

import constants.HttpConstant;
import model.form.*;
import model.http.HttpRequestParam;
import org.apache.commons.lang3.StringUtils;
import util.HttpUtil;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import static constants.HttpConstant.FORM_DATA_BASE_URL;
import static constants.HttpConstant.FORM_DATA_V1_URL;

/**
 * 表单数据相关的请求Demo
 */
public class FormDataDemo {


    /**
     * 新建单条数据接口
     *
     * @param createParam
     * @throws Exception
     */
    public Map<String, Object> singleDataCreate(FormDataCreateParam createParam) throws Exception {
        if (StringUtils.isBlank(createParam.getAppId()) || StringUtils.isBlank(createParam.getEntryId())) {
            throw new RuntimeException("param lack!");
        }
        String url = FORM_DATA_BASE_URL + createParam.getAppId() + "/entry/" + createParam.getEntryId() + "/data_create";
        HttpRequestParam param = new HttpRequestParam();
        param.setApiKey(HttpConstant.API_KEY);
        param.setUrl(url);
        Map<String, Object> data = new HashMap<>();
        data.put("data", createParam.getData());
        data.put("is_start_workflow", createParam.getIs_start_workflow());
        data.put("is_start_trigger", createParam.getIs_start_trigger());
        data.put("transaction_id", createParam.getTransaction_id());
        param.setData(data);
        Map<String, Object> result = HttpUtil.sendPostRequest(param);
        return result;
    }


    /**
     * 查询单条数据接口
     *
     * @param appId
     * @param entryId
     * @param dataId
     * @return
     * @throws Exception
     */
    public Map<String, Object> singleDataQuery(String appId, String entryId, String dataId) throws Exception {
        if (StringUtils.isBlank(appId) || StringUtils.isBlank(entryId)) {
            throw new RuntimeException("param lack!");
        }
        String url = FORM_DATA_BASE_URL + appId + "/entry/" + entryId + "/data_retrieve";
        HttpRequestParam param = new HttpRequestParam();
        param.setApiKey(HttpConstant.API_KEY);
        param.setUrl(url);
        Map<String, Object> data = new HashMap<>();
        data.put("data_id", dataId);
        param.setData(data);
        Map<String, Object> result = HttpUtil.sendPostRequest(param);
        return result;
    }

    /**
     * 修改单条数据接口
     *
     * @param updateParam
     * @return
     * @throws Exception
     */
    public Map<String, Object> singleDataUpdate(FormDataUpdateParam updateParam) throws Exception {
        if (StringUtils.isBlank(updateParam.getAppId()) || StringUtils.isBlank(updateParam.getEntryId())
                || StringUtils.isBlank(updateParam.getDataId())) {
            throw new RuntimeException("param lack!");
        }
        String url = FORM_DATA_BASE_URL + updateParam.getAppId() + "/entry/" + updateParam.getEntryId() + "/data_update";
        HttpRequestParam param = new HttpRequestParam();
        param.setApiKey(HttpConstant.API_KEY);
        param.setUrl(url);
        Map<String, Object> data = new HashMap<>();
        data.put("data_id", updateParam.getDataId());
        data.put("data", updateParam.getData());
        data.put("is_start_trigger", updateParam.getIs_start_trigger());
        data.put("transaction_id", updateParam.getTransaction_id());
        param.setData(data);
        Map<String, Object> result = HttpUtil.sendPostRequest(param);
        return result;
    }


    /**
     * 刪除单条数据接口
     *
     * @param deleteParam
     * @return
     * @throws Exception
     */
    public Map<String, Object> singleDataRemove(FormDataDeleteParam deleteParam) throws Exception {
        if (StringUtils.isBlank(deleteParam.getAppId()) || StringUtils.isBlank(deleteParam.getDataId()) ||
                StringUtils.isBlank(deleteParam.getEntryId())) {
            throw new RuntimeException("param lack!");
        }
        String url = FORM_DATA_V1_URL + deleteParam.getAppId() + "/entry/" + deleteParam.getEntryId() + "/data_delete";
        HttpRequestParam param = new HttpRequestParam();
        param.setApiKey(HttpConstant.API_KEY);
        param.setUrl(url);
        Map<String, Object> data = new HashMap<>();
        data.put("data_id", deleteParam.getDataId());
        data.put("is_start_trigger", deleteParam.getIs_start_trigger());
        param.setData(data);
        Map<String, Object> result = HttpUtil.sendPostRequest(param);
        return result;
    }

    /**
     * 新建多条数据接口
     *
     * @param createParam
     * @throws Exception
     */
    public Map<String, Object> batchDataCreate(FormDataBatchCreateParam createParam) throws Exception {
        if (StringUtils.isBlank(createParam.getAppId()) || StringUtils.isBlank(createParam.getEntryId())) {
            throw new RuntimeException("param lack!");
        }
        String url = FORM_DATA_V1_URL + createParam.getAppId() + "/entry/" + createParam.getEntryId() + "/data_batch_create";
        HttpRequestParam param = new HttpRequestParam();
        param.setApiKey(HttpConstant.API_KEY);
        param.setUrl(url);
        Map<String, Object> data = new HashMap<>();
        data.put("data_list", createParam.getDataList());
        data.put("is_start_workflow", createParam.getIs_start_workflow());
        data.put("transaction_id", createParam.getTransaction_id());
        param.setData(data);
        Map<String, Object> result = HttpUtil.sendPostRequest(param);
        return result;
    }

    /**
     * 查询多条数据接口
     *
     * @param queryParam
     * @throws Exception
     */
    public Map<String, Object> batchDataQuery(FormDataQueryParam queryParam) throws Exception {
        if (StringUtils.isBlank(queryParam.getAppId()) || StringUtils.isBlank(queryParam.getEntryId())) {
            throw new RuntimeException("param lack!");
        }
        String url = FORM_DATA_V1_URL + queryParam.getAppId() + "/entry/" + queryParam.getEntryId() + "/data";
        HttpRequestParam param = new HttpRequestParam();
        param.setApiKey(HttpConstant.API_KEY);
        param.setUrl(url);
        Map<String, Object> data = new HashMap<>();
        data.put("data_id", queryParam.getDataId());
        data.put("fields", queryParam.getFieldList());
        data.put("filter", queryParam.getFilter());
        data.put("limit", queryParam.getLimit());
        param.setData(data);
        Map<String, Object> result = HttpUtil.sendPostRequest(param);
        return result;
    }

    /**
     * 删除多条数据接口
     *
     * @param appId
     * @param entryId
     * @param dataIdList
     * @return
     * @throws Exception
     */
    public Map<String, Object> batchDataRemove(String appId, String entryId, List<String> dataIdList) throws Exception {
        if (StringUtils.isBlank(appId) || StringUtils.isBlank(entryId)) {
            throw new RuntimeException("param lack!");
        }
        String url = FORM_DATA_V1_URL + appId + "/entry/" + entryId + "/data_batch_delete";
        HttpRequestParam param = new HttpRequestParam();
        param.setApiKey(HttpConstant.API_KEY);
        param.setUrl(url);
        Map<String, Object> data = new HashMap<>();
        data.put("data_ids", dataIdList);
        param.setData(data);
        Map<String, Object> result = HttpUtil.sendPostRequest(param);
        return result;
    }

    /**
     * 修改多条数据接口
     *
     * @param updateParam
     * @return
     * @throws Exception
     */
    public Map<String, Object> batchDataUpdate(FormDataBatchUpdateParam updateParam) throws Exception {
        if (StringUtils.isBlank(updateParam.getAppId()) || StringUtils.isBlank(updateParam.getEntryId())) {
            throw new RuntimeException("param lack!");
        }
        String url = FORM_DATA_V1_URL + updateParam.getAppId() + "/entry/" + updateParam.getEntryId() + "/data_batch_update";
        HttpRequestParam param = new HttpRequestParam();
        param.setApiKey(HttpConstant.API_KEY);
        param.setUrl(url);
        Map<String, Object> data = new HashMap<>();
        data.put("data_ids", updateParam.getDataIds());
        data.put("transaction_id", updateParam.getTransaction_id());
        data.put("data", updateParam.getData());
        param.setData(data);
        Map<String, Object> result = HttpUtil.sendPostRequest(param);
        return result;
    }

//
//    /**
//     * 修改多条数据接口
//     */
//    async batchDataUpdate(appId, entryId, dataIds, data, options) {
//        return await this.doRequest({
//                version: 'v1',
//                method: 'POST',
//                path: `/app/${appId}/entry/${entryId}/data_batch_update`,
//                payload: {
//            data_ids: dataIds,
//                    data,
//                    transaction_id: options.transactionId
//        }
//        });
//    }


}

File: python/src/api/jdy/form_data_demo.py
from ...util import http_util as http_util
from ...constants.http_constant import HttpConstant
from ...model.http.http_request_param import HttpRequestParam

"""
新建单条数据接口
Arguments:
    param: FormDataCreateParam 实例
"""


def singleDataCreate(param):
    url = HttpConstant.FORM_DATA_BASE_URL + param.app_id + "/entry/" + param.entry_id + "/data_create"
    request_param = HttpRequestParam(HttpConstant.API_KEY, url, param)
    return http_util.send_post(request_param)


"""
查询单条数据接口
Arguments:
    appId: str 应用id
    entry_id: str 表单id
    data_id: str 数据id
"""


def singleDataQuery(app_id, entry_id, data_id):
    url = HttpConstant.FORM_DATA_BASE_URL + app_id + "/entry/" + entry_id + "/data_retrieve"
    request_param = HttpRequestParam(HttpConstant.API_KEY, url, {'data_id': data_id})
    return http_util.send_post(request_param)


"""
修改单条数据接口
Arguments:
    param: FormDataUpdateParam 实例
"""


def singleDataUpdate(param):
    url = HttpConstant.FORM_DATA_BASE_URL + param.app_id + "/entry/" + param.entry_id + "/data_update"
    request_param = HttpRequestParam(HttpConstant.API_KEY, url, param)
    return http_util.send_post(request_param)


"""
刪除单条数据接口
Arguments:
    param: FormDataDeleteParam 实例
"""


def singleDataRemove(param):
    url = HttpConstant.APP_BASE_URL + param.app_id + "/entry/" + param.entry_id + "/data_delete"
    request_param = HttpRequestParam(HttpConstant.API_KEY, url, param)
    return http_util.send_post(request_param)


"""
新建多条数据接口
Arguments:
    param: FormDataBatchCreateParam 实例
"""


def batchDataCreate(param):
    url = HttpConstant.APP_BASE_URL + param.app_id + "/entry/" + param.entry_id + "/data_batch_create"
    request_param = HttpRequestParam(HttpConstant.API_KEY, url, param)
    return http_util.send_post(request_param)


"""
查询多条数据接口
Arguments:
    param: FormDataQueryParam 实例
"""


def batchDataQuery(param):
    url = HttpConstant.APP_BASE_URL + param.app_id + "/entry/" + param.entry_id + "/data"
    request_param = HttpRequestParam(HttpConstant.API_KEY, url, param)
    return http_util.send_post(request_param)


"""
删除多条数据接口
Arguments:
    appId: str 应用id
    entry_id: str 表单id
    data_ids: list<str> 数据id list
"""


def batchDataRemove(app_id, entry_id, data_ids):
    url = HttpConstant.APP_BASE_URL + app_id + "/entry/" + entry_id + "/data_batch_delete"
    request_param = HttpRequestParam(HttpConstant.API_KEY, url, {'data_ids': data_ids})
    return http_util.send_post(request_param)


"""
修改多条数据接口
Arguments:
    param:FormDataBatchUpdateParam 实例
"""


def batchDataUpdate(param):
    url = HttpConstant.APP_BASE_URL + param.app_id + "/entry/" + param.entry_id + "/data_batch_update"
    request_param = HttpRequestParam(HttpConstant.API_KEY, url, param)
    return http_util.send_post(request_param)



